examples/audio_server/server.cc

```diff
@@ -30,101 +31,113 @@ int main(int argc, char** argv) {
     auto& http = HTTP(current::net::AcquireLocalPort(FLAGS_port));
 
     current::WaitableAtomic<SharedState> safe_state;
-    std::map<std::string, std::thread> threads;
 
     // Need to join and remove finished sessions
-    auto joiner = std::thread([&threads, &safe_state]{
+    auto joiner = std::thread([&safe_state]{
       while(true) {
-        std::vector<std::string> to_join;
-        for (auto& pair : threads) {
-          std::string channel_id = pair.first;
-          auto has_channel = safe_state.MutableUse([channel_id](SharedState& state) {
-            return state.channel_exists(channel_id);
-          });
-          if (!has_channel) {
-            to_join.push_back(pair.first);
-            pair.second.join();
+        auto die = safe_state.MutableUse([](SharedState& state){
+          if(state.die){
+            return true;
           }
-        }
-        for(auto& to_drop: to_join){
-          threads.erase(to_drop);
+          std::vector<std::string> to_join;
+          for (auto& pair : state.threads) {
+            std::string channel_id = pair.first;
+            if (!state.channel_exists(channel_id)) {
+              to_join.push_back(pair.first);
+              pair.second.join();
+            }
+          }
+          for(auto& to_drop: to_join){
+            state.threads.erase(to_drop);
+          }
+          return false;
+        });
+        if(die){
+          break;
         }
         std::this_thread::sleep_for(std::chrono::milliseconds(1000));
       }
     });
 
-    auto scope = http.Register("/", [&safe_state, &threads](Request r) {
+    auto scope = http.Register("/", [&safe_state](Request r) {
       std::string channel_id = r.body;
+      // TODO: do not forget to pass input/output port for channel creation handler
 
-      auto has_channel = safe_state.MutableUse([channel_id](SharedState& state) { return state.channel_exists(channel_id); });
-      if(has_channel){
-        r("error: channel already exists\n");
-        return;
-      }
-      uint32_t n_chans = safe_state.MutableUse([](SharedState& state) { return state.active_channels; });
-      if(n_chans == FLAGS_n){
-        r("error: too many channels\n");
-        return;
-      }
+      struct channel_validation{
```

Comment:
Btw, if this one is a `CURRENT_STRUCT` (declared outside any function), you could just respond with it from the HTTP handler, and the caller will get the JSON.